Decorator.cpp

```diff
@@ -0,0 +1,100 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+class Burger
+{
+public:
+    virtual string Serve() = 0;
+    virtual float price() = 0;
+};
+
+
+class ClassicBurger : public Burger
+{
+public:
+    string Serve()
+    {
+        return "Burger";
+    }
+
+    float price()
+    {
+        return 40;
+    }
+};
+
+class BurgerDecorator: public Burger
+{
+protected:
+    Burger *m_Burger;
+public:
+
+    BurgerDecorator(Burger *classicBurger): m_Burger(classicBurger){}
+
+    string Serve()
+    {
+        return m_Burger->Serve();
+    }
+
+    float price()
+    {
+        return m_Burger ->price();
+    }
+};
+
+
+class DoubleCheeseBurger: public BurgerDecorator
+{
+public:
+    DoubleCheeseBurger(Burger *classicBurger): BurgerDecorator(classicBurger){}
+
+    string Serve()
+    {
+        return m_Burger->Serve() + " decorated with DOUBLE CHEESE ";
+    }
+    float price()
+    {
+        return m_Burger ->price() + 40;
```

Comment:
it would be more readable if you defined 40 as the price of a double cheeseburger.
You can do it for all classes.